### ISSUE TITLE
sometimes no clock offsets get recorded

### ISSUE THREAD

**Issue by tstenner** [labstreaminglayer/liblsl#8]
See https://github.com/sccn/labstreaminglayer/issues/338

**Comment by tstenner** [comment]
Current status of the investigation: on at least one lab computer the inlet tried to connect to the data transfer port, which is the same as the clock synchronization port most of the time but not always.

**Comment by tstenner** [comment]
Closed in 5977990cf8bf2d88976cfd630dfdc54c313f6fe6

**Comment by mgrivich** [comment]
In parallel, both Tristan and I diagnosed and corrected the bug, and there is a new release at https://github.com/sccn/liblsl/releases/tag/1.13-b2. You'll notice that release is beta. We are looking to do a bit more testing before making an official 1.13. I wrote the original bug. :/. I'm very sorry, considering how many people were impacted by this.

The bug was introduced in ff1e8e1a269d0848fdf1bbe98c8bc3ddf6d6f35b, 2017-07-13. While fixing an issue with ipv6 addresses, I accidentally set udp to look at the data_port rather than the service_port. Often these are the same, but sometimes they are not, which caused the udp port to fail intermittently. However, it could be reliably reproduced if you did the exact same action every time (in my case, opening the receive string before send string, and checking for timestamps on the receiver).
